app/models/capability.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Capability
```

Comment:
Obviously similar comments apply here as with `Ability` but this one I think is something of your invention as I don't see it mentioned in the documentation?

I wonder if this is even needed - given we are overriding `current_ability` could we not pass both the user and token to `Ability` and have it do everything?

Reply:
I don't have any views on this, beyond what @cflipse wrote in the [commit message](https://github.com/openstreetmap/openstreetmap-website/pull/2023/commits/91fc65a2e3ad47414344a6d5dc03cb5fb2a08ce1) when he split them out.

To be blunt, I don't yet have a good understanding of how all the tokens stuff works, so I'm happy to take direction here. My experience elsewhere is just with straightforward approaches around having a current_user with particular roles, and our token handling here is more complex.

Reply:
It's been a while.  IIRC, the capabilities reflect permissions granted to the application -- so, if it's making a request to access your GPS, and you say "yes", then you've granted the app that capability.  Another common example is when you OAuth login and the system asks for permission to read your contact lists.

This is, more or less, inverse of the CanCanCan's normal idea of an Ability, which is the app deciding if the user has permission to do something; Capability is the user deciding if the app has permission to do something.  (Capability was an inherited name, I suspect that something better could be determined)  The end result of the calculation is the same, but separating them helps to keep them from getting too crossed.